poppy/poppy_core.py

```diff
@@ -687,16 +699,31 @@ def _resample_wavefront_pixelscale(self, detector):
         _log.debug("Desired detector FOV: {} pixels, {:.3f}".format(detector.shape,
                                                                     detector.shape[0]*u.pixel*detector.pixelscale))
 
-        # TODO the following works but is not optimal for performance
-        # We should consider cropping out an appropriate subregion prior to performing the zoom.
-        # That makes a difference if the detector is only sampling a small part of a much larger wavefront
-
-        new_wf_real = scipy.ndimage.zoom(self.wavefront.real, pixscale_ratio, order=detector.interp_order)
-        new_wf_imag = scipy.ndimage.zoom(self.wavefront.imag, pixscale_ratio, order=detector.interp_order)
-        new_wf = new_wf_real + 1.j*new_wf_imag
+        def make_axis(npix, step):
+            """ Helper function to make coordinate axis for interpolation """
+            return step * np.arange(-npix // 2, npix // 2, dtype=np.float64)
+
+        # Provide 2x margin around image to reduce interpolation errors at edge, but also make
```

Comment:
Can  you explain this 2x margin factor more? That seems a somewhat blunt approach, and I expect would make the interpolation on the order of 4x slower?

Reply:
We definitely want to provide some margin around the image, so I chose a small integer multiple to guarantee that there wouldn't be any issues with PSF centering after cropping.

This can be made much smaller, as long as the margin is an integer number of pixels on all sides.  I don't have a good sense of how large the margin needs to be to minimize interpolation edge artifacts- maybe a small number of pixels is enough.